backend/src/main/java/com/ody/route/domain/ClientType.java

```diff
@@ -1,8 +1,21 @@
 package com.ody.route.domain;
 
+import java.time.LocalDate;
+import java.util.function.Function;
+
 public enum ClientType {
 
-    ODSAY,
-    GOOGLE,
+    ODSAY(date -> date),
+    GOOGLE(date -> date.withDayOfMonth(1)),
     ;
+
+    private final Function<LocalDate, LocalDate> resetStrategy;
```

Comment:
[제안] 입출력이 같은 경우, `UnaryOperator` 사용할 수 있다고 하네요!
```suggestion
    private final UnaryOperator<LocalDate> resetStrategy;
```

Reply:
[반영] 새롭게 배웠습니다!! [e829021](https://github.com/woowacourse-teams/2024-ody/pull/732/commits/e8290217377d92ee08b9dbb7021c4ad38bfd65f9)